### PR TITLE
feat: auto id pattern

### PR DESCRIPTION
**Pull request by fdlk**
Allows you to tag an AUTO ID STRING attribute to generate incrementing string IDs.
In tag with relation IRI [obo:IAO_0000596](http://www.ontobee.org/ontology/IAO?iri=http://purl.obolibrary.org/obo/IAO_0000596) you specify the number of digits, e.g. `7`
In the other tag with relation IRI [obo:IAO_0000599](http://www.ontobee.org/ontology/IAO?iri=http://purl.obolibrary.org/obo/IAO_0000599) you specify the ID prefix, e.g. `GEN-`

Example of tags:
| identifier       | label | relationLabel   | relationIRI                                |
|------------------|-------|-----------------|--------------------------------------------|
| hasIDDigitCount7 | 7     | hasIDDigitCount | http://purl.obolibrary.org/obo/IAO_0000596 |
| hasIDPrefixGen   | GEN-  | hasIDPrefix     | http://purl.obolibrary.org/obo/IAO_0000599 |

Example of how to use these tags in the auto ID attribute:
| name      | dataType | nillable | idAttribute | tags                            |
|-----------|----------|----------|-------------|---------------------------------|
| pseudonym | string   | false    | AUTO        | hasIDDigitCount7,hasIDPrefixGen |

[PR-9332.xlsx](https://github.com/molgenis/molgenis/files/7449857/PR-9332.xlsx)

Check out the [preview](https://preview-pr-9332-7.dev.molgenis.org/menu/main/dataexplorer?entity=fdlk_pseudonym&hideselect=true)

Solved issues:
* scramble feature -> moved to #9343 
* predictability -> now super predictable 1,2,3
* One sequence per attribute -> done
* Roundtrip emx -> still a bit hacky, but probably won't happen that much.

I've added a low-level endpoint to the metadata service that allows you to query sequence names and set a correct value for a sequence after import
See https://github.com/molgenis/molgenis/blob/47cb44fa5d2113533af3fc0a2ff9162b6d37926d/molgenis-metadata-manager/src/test/resources/test-sequences.http for examples of how to retrieve the names of the sequences and how to set a new value for a sequence. (make sure you url encode the # in the sequence name properly)

#### Checklist
- [ ] Functionality works & meets specifications
- [x] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] Migration step added in case of breaking change
- [ ] User documentation updated
- [ ] (If you have changed REST API interface) view-swagger.ftl updated
- [ ] Test plan template updated
- [x] Clean commits
- [ ] Added Feature/Fix to release notes

### CODE REVIEW

File: molgenis-data-postgresql/src/main/java/org/molgenis/data/postgresql/JdbcSequenceIdGenerator.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  @PostConstruct
+  private void init() {
+    jdbcOperations.execute(format("CREATE SEQUENCE IF NOT EXISTS %s", sequenceName));
+  }
+
+  @Override
+  public int generateId() {
+    var result = jdbcOperations.queryForMap("select nextval(?)", sequenceName);
+    return ((Long) result.get("nextval")).intValue();
```
Comment (fdlk) [inline molgenis-data-postgresql/src/main/java/org/molgenis/data/postgresql/JdbcSequenceIdGenerator.java:27]: why not longs and cut them down later?

File: molgenis-data-postgresql/src/main/java/org/molgenis/data/postgresql/SequenceIdConfig.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import javax.sql.DataSource;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@Configuration
+public class SequenceIdConfig {
+
+  public static final String DEFAULT_ID_SEQ = "default_id_seq";
```
Comment (fdlk) [inline molgenis-data-postgresql/src/main/java/org/molgenis/data/postgresql/SequenceIdConfig.java:12]: make specific for the id attribute

**Comment by tommydeboer** [comment]
[Merge pseudonimisation PRs](https://trello.com/c/HDzlGGav/394-merge-pseudonimisation-prs)

**Comment by tommydeboer** [comment]
Continued in #9346

**Pull request closed without merging**